django_setup_configuration/model_utils.py

```diff
@@ -22,6 +23,58 @@
 )
 
 
+class YamlWithEnvSubstitution(YamlConfigSettingsSource):
+    """Modified YAML source that recursively substitutes markers with env vars."""
+
+    def __init__(self, namespace: str, **kwargs):
+        self.namespace = namespace
+        super().__init__(**kwargs)
+
+    @staticmethod
+    def substitute(field: dict | list, field_name: str):
```

Comment:
When recursing, don't we reach the "base values" like str/number/bool at some point?